### PR TITLE
MHV-57332 SM signature for ROI triage teams

### PR DESCRIPTION
**Pull request by alex-morgun-1**
## Summary

**User Story:**  As a SM user, I want to add a legal signature to my message when I am messaging any of the 3 medical records triage groups.

**Medical record triage group flow**

**Given**: a user is on the "start a new message" screen
**when**: the user selects a medical records triage group
**then**: digital signature (state of truth) component will appear
**and**: an alert will appear with focus on the dismiss "Messages to this team require a signature. We added a signature box to this page."

 
**User starts with MR TG --> adds text to signature field --> then switches to non-MR TG flow**

**Given**: a user is on the "start a new message" screen
**And**: the user has selected a medical records triage group which displays the digital signature (statement of truth) component
**And**: the user has entered text into the digital signature component

**When**: the user changes to a non-medical records triage group
**Then**: an alert will appear with focus on the dismiss "Messages to this team don't require a signature. We removed the signature box from this page."
**and**: the signature (statement of truth) component is removed from the page

 

**User starts with MR TG --> NO text to signature field --> then switches to non-MR TG Flow**

**Given**: a user is on the "start a new message" screen
**And**: the user has selected a medical records triage group which displays the digital signature (statement of truth) component
**and**: the user has not entered text into the signature component
**then**: the user switches the recipient to a non-medical record triage group
**then**: the signature component is removed from the page
**and**: a screen-reader only alert will read off "Messages to this team don't require a signature. We removed the signature box from this page."

Error Flows: https://www.figma.com/design/x9YoKDPekIPiqlDiZdmjci/ZFeat---Changes-to-signature%2Fpreferences?node-id=682-10912&t=VMwrbna2FgTJbdiV-4

User Story:  As a SM user, I want to see error messages when I try to save a draft with a signature or a signature with an attachment, so that I understand I can't save as a draft with these components.

Given A user has started a new message with a signature
When A user tries to save a draft message with a signature{}
Then A modal will appear alerting the user that messages can't be saved with a signature

Given A user has started a new message with a signature and an attachment
When  A user tries to save a draft message with a signature and an attachment{}
Then A modal will appear alerting the user that messages can't be saved with a signature or an attachment

## Related issue(s)

- https://jira.devops.va.gov/browse/MHV-57332
- https://jira.devops.va.gov/browse/MHV-58957

## Testing done

- _Describe what the old behavior was prior to the change_
- _Describe the steps required to verify your changes are working as expected_
- _Describe the tests completed and the results_
- _Exclusively stating 'Specs and automated tests passing' is NOT acceptable as appropriate testing
- _Optionally, provide a link to your [test plan](https://depo-platform-documentation.scrollhelp.site/developer-docs/create-a-test-plan-in-testrail) and [test execution records](https://depo-platform-documentation.scrollhelp.site/developer-docs/execute-tests-in-testrail)_

## Screenshots

_Note: This field is mandatory for UI changes (non-component work should NOT have screenshots)._

|         | Before | After |
| ------- | ------ | ----- |
| Mobile  |image image|image image|
| Desktop |imageimage|image image image image image|

## What areas of the site does it impact?

*(Describe what parts of the site are impacted **if** code touched other areas)*

### Quality Assurance & Testing

- [ ] I fixed|updated|added unit tests and integration tests for each feature (if applicable).
- [ ] No sensitive information (i.e. PII/credentials/internal URLs/etc.) is captured in logging, hardcoded, or specs
- [ ] Linting warnings have been addressed
- [ ] Documentation has been updated ([link to documentation](#) \*if necessary)
- [ ] Screenshot of the developed feature is added
- [ ] [Accessibility testing](https://depo-platform-documentation.scrollhelp.site/developer-docs/wcag-2-1-success-criteria-and-foundational-testing) has been performed

### Error Handling

- [ ] Browser console contains no warnings or errors.
- [ ] Events are being sent to the appropriate logging solution
- [ ] Feature/bug has a monitor built into Datadog or Grafana (if applicable)

### Authentication

- [ ] Did you login to a local build and verify all authenticated routes work as expected with a test user

### :warning: Team Sites (only applies to modifications made to the VA.gov header) :warning:

- [ ] The vets-website header does not contain any web-components
- [ ] I used the [proxy-rewrite steps](https://github.com/department-of-veterans-affairs/vets-website/tree/main/src/applications/proxy-rewrite#local-dev) to test the injected header scenario
- [ ] I reached out in the `#sitewide-public-websites` Slack channel for questions

## Requested Feedback

(OPTIONAL) _What should the reviewers know in addition to the above. Is there anything specific you wish the reviewer to assist with. Do you have any concerns with this PR, why?_

### CODE REVIEW

File: src/applications/mhv-secure-messaging/components/ComposeForm/RecipientsSelect.jsx [hunk 1]
```diff
@@ -0,0 +1,144 @@
+/**
+ * Renders a select dropdown for selecting recipients in the compose form.
+ *  - If a recipient requires a signature or switching from a recipient that does not 
+ *    require a signature to one that does, an alert is displayed notifyong that signature is required.
```
Comment (KyleCardwell) [inline src/applications/mhv-secure-messaging/components/ComposeForm/RecipientsSelect.jsx:4]: small typo - notifying

**Review by KyleCardwell — DISMISSED** [review]
a lot of great work!  looks good.

File: src/applications/mhv-secure-messaging/components/ComposeForm/RecipientsSelect.jsx (hunk 1 above)
Comment (oleksii-morgun) [reply src/applications/mhv-secure-messaging/components/ComposeForm/RecipientsSelect.jsx:4]: thank you, will make a correction 👍🏻

**Review by KyleCardwell — APPROVED** [review]
LGTM

**Pull request merged**